pkg/apis/componentconfig/v1alpha1/defaults.go

```diff
@@ -129,18 +129,29 @@ func SetDefaults_KubeSchedulerConfiguration(obj *KubeSchedulerConfiguration) {
 	}
 
 	if len(obj.AlgorithmSource.Type) == 0 {
-		obj.AlgorithmSource.Type = SchedulerAlgorithmSourcePolicy
+		obj.AlgorithmSource.Type = SchedulerAlgorithmSourceProvider
 	}
-	if obj.AlgorithmSource.Policy != nil &&
-		obj.AlgorithmSource.Policy.ConfigMap != nil &&
-		len(obj.AlgorithmSource.Policy.ConfigMap.Reference.Namespace) == 0 {
-		obj.AlgorithmSource.Policy.ConfigMap.Reference.Namespace = api.NamespaceSystem
-	}
-	if obj.AlgorithmSource.Type == SchedulerAlgorithmSourceProvider {
+
+	switch obj.AlgorithmSource.Type {
+	case SchedulerAlgorithmSourceProvider:
 		if obj.AlgorithmSource.Provider == nil || len(*obj.AlgorithmSource.Provider) == 0 {
 			val := SchedulerDefaultProviderName
 			obj.AlgorithmSource.Provider = &val
 		}
+	case SchedulerAlgorithmSourcePolicy:
+		// There is no default file or config map to apply.
+	}
+
+	if obj.AlgorithmSource.Policy != nil {
+		switch obj.AlgorithmSource.Policy.Type {
```

Comment:
Why do we need this switch case? All that is happening here is to set a default namespace for a provided ConfigMap when it doesn't have a namespace.